### PR TITLE
[dong] bug p2p #858

### PR DESCRIPTION
**Pull request by phongvu2812**
# Ticket
#858

# General
- [ ] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [ ] I fully understand the code I submitted.
- [ ] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

**Review by vietdangquoc — APPROVED** [review]

File: frontend/modules/p2p/components/p2p-trading/create-offer-form/payment-section.tsx [hunk 1]
```diff
@@ -50,26 +57,20 @@ export const PaymentSection = ({ control, setValue, watch, onUnsavedChangesChang
...
-        }
+    // Synchronize payment_info_id if we have a strict match but no ID or wrong ID set
+    if (matchedSavedInfo) {
+      if (watch('payment_info_id') !== matchedSavedInfo.id) {
+        setValue('payment_info_id', matchedSavedInfo.id, { shouldValidate: true });
+      }
+    } else {
+      // If no strict match (info differs from saved record), clear the ID
+      if (watch('payment_info_id')) {
+        setValue('payment_info_id', undefined, { shouldValidate: true });
```
Comment (hoangdohuy) [inline frontend/modules/p2p/components/p2p-trading/create-offer-form/payment-section.tsx:102]: check logic lúc mới đầu vào nếu chưa selection gì thì default seleection là gì cho cả ở màn create offer và màn confirm buy ở order

**Review by hoangdohuy — APPROVED** [review]
LGTM
